components/nifosg/nifloader.cpp

```diff
@@ -1030,6 +1030,78 @@ namespace NifOsg
             }
         }
 
+        class PreciseLeafDepthCullCallback : public osg::DrawableCullCallback{
+            inline osgUtil::CullVisitor::value_type distance(const osg::Vec3& coord,const osg::Matrix& matrix) const
+            {
+                return -((osgUtil::CullVisitor::value_type)coord[0]*(osgUtil::CullVisitor::value_type)matrix(0,2)+(osgUtil::CullVisitor::value_type)coord[1]*(osgUtil::CullVisitor::value_type)matrix(1,2)+(osgUtil::CullVisitor::value_type)coord[2]*(osgUtil::CullVisitor::value_type)matrix(2,2)+matrix(3,2));
+            }
+
+            virtual bool cull(osg::NodeVisitor* nv, osg::Drawable* drawable, osg::RenderInfo* renderInfo) const{
+                osgUtil::CullVisitor * cv = nv -> asCullVisitor();
```

Comment:
`osg::NodeVisitor::asCullVisitor` didn't exist until OSG 3.5.2, so this won't work on 3.4.1. Your options are:

* A `dynamic_cast` - slowness that isn't necessary.
* Checking the type of visitor with `osg::NodeVisitor::getVisitorType() == osg::NodeVisitor::CULL_VISITOR` and then a `static_cast` if it's true - less slowness, but unnecessary if you know it's always going to be a cull visitor.
* Just doing a `static_cast` - if you can guarantee that cull callbacks only get called by cull visitors (which seems sensible) a static cast is fine. It'll also be faster than the virtual function call you've already got here.
* Checking the OSG version with a macro and using a different function depending on that - probably unnecessary and overcomplicated, but might avoid using a slow path on 3.6.x just becasue 3.4.x didn't have the fast path yet.

Reply:
static_cast is cool